dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java

```diff
@@ -244,4 +297,633 @@ public final Response switchSite(
+                .sorted(hostNameComparator).map(host -> this.toSiteMap(user, contentletAPI, host))
+                .collect(Collectors.toList()))).build();
+    }
+
+    private Map<String, Object> toSiteMap(final User user,
+                                          final ContentletAPI contentletAPI, final Host host) {
+
+        final Map<String, Object> thumbInfo = new HashMap<>();
+        thumbInfo.put("hostId",    host.getIdentifier());
+        thumbInfo.put("hostInode", host.getInode());
+        thumbInfo.put("hostName",  host.getHostname());
+        final File hostThumbnail   = Try.of(()-> contentletAPI.getBinaryFile(host.getInode(), Host.HOST_THUMB_KEY, user)).getOrNull();
+        final boolean hasThumbnail = hostThumbnail != null;
+        thumbInfo.put("hasThumbnail", hasThumbnail);
+        thumbInfo.put("tagStorage",   host.getMap().get("tagStorage"));
+        return thumbInfo;
+    }
+
+    /**
+     * Publish a host
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @PUT
+    @Path("/_publish/{hostId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response publishSite(@Context final HttpServletRequest httpServletRequest,
+                                @Context final HttpServletResponse httpServletResponse,
+                                @PathParam("hostId") final String hostId) throws DotDataException, DotSecurityException, PortalException, SystemException {
+
+        final User user = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(httpServletRequest, httpServletResponse)
+                .requiredBackendUser(true)
+                .rejectWhenNoUser(true)
+                .requireLicense(true)
+                .init().getUser();
+
+        Logger.debug(this, ()-> "Publishing site: " + hostId);
+
+        final PageMode      pageMode      = PageMode.get(httpServletRequest);
+        final Host host = pageMode.respectAnonPerms? this.siteHelper.getSite(user, hostId):
+                this.siteHelper.getSiteNoFrontEndRoles(user, hostId);
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+        this.siteHelper.publish(host, user, pageMode.respectAnonPerms);
+        return Response.ok(new ResponseEntityView(this.toView(host))).build();
+    }
+
+    /**
+     * Unpublish a host
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @PUT
+    @Path("/_unpublish/{hostId}")
```

Comment:
I think it's better to have the hostId and then the action the exec. `/{hostId}/_unpublish`

Reply:
Agreed